src/server/controllers/Grbl/GrblController.js

```diff
@@ -1345,6 +1346,20 @@ class GrblController {
 
                     this.command('gcode:load', file, data, context, callback);
                 });
+            },
+            'jogCancel': () => {
+                // The GRBL jog cancel command (0x85) is a realtime command, so it will
+                // be handled by GRBL as soon as it enters the serial buffer. If there
+                // are any unparsed jog commands ($J=) in GRBL's serial buffer the
+                // cancel command will skip ahead of them and cancel any parsed jog
+                // commands, but then the unparsed jog command will be parsed and jog
+                // one more time.  Wait until all the commands sent by the feeder have
+                // been acked to ensure all pending jog commands have been parsed
+                // before sending the jog cancel command.
+                let [context = {}] = args;
+                this.feeder.onEmpty((context) => {
+                    this.command('gcode', '\x85', context);
+                }, context);
```

Comment:
I have a question about the change because the "onEmpty" callback might be unnecessary here.

The feeder itself is using a so-called **SEND-RESPONSE** approach, it will send the next line to the serial buffer only when it receives an "ok" or "error" response from Grbl.

So this function can be simplified as below by putting `\x85` to the end of the feeder queue:

```js
'jogCancel': () => {
  this.command('gcode', '\x85', context);
},
```

Reply:
If necessary, I can add support for prepending data to the Feeder queue rather than appending for high priority commands.

Reply:
It's been a long time since I looked at this, but if I remember correctly just adding the jogCancel command at the end of the queue didn't work.  I can try to find time to play with it some more.  I don't think prepending will work either, the necessary semantics is that jogCancel needs to stop all jog commands that have been sent before it.  If you prepend the jogCancel command before some pending jog commands then the movement will stop and then restart.

Reply:
Colin any chance you can make a compiled build for windows available that includes this change?